auth/api/iam/api_test.go

```diff
@@ -847,13 +848,56 @@ func TestWrapper_StatusList(t *testing.T) {
 		ctx := newTestClient(t)
 		ctx.vcIssuer.EXPECT().StatusList(nil, gomock.Any(), gomock.Any()).Return(nil, types.ErrNotFound)
 
-		res, err := ctx.client.StatusList(nil, StatusListRequestObject{})
+		res, err := ctx.client.StatusList(nil, StatusListRequestObject{Did: webDID.String()})
 
 		assert.ErrorIs(t, err, types.ErrNotFound)
 		assert.Nil(t, res)
 	})
 }
 
+func Test_createOAuth2EndpointURL(t *testing.T) {
+	t.Run("no endpoint", func(t *testing.T) {
+		webDID := did.MustParseDID("did:web:example.com:iam:holder")
+		actual, err := createOAuth2EndpointURL(webDID)
+
+		require.NoError(t, err)
+		require.NotNil(t, actual)
+		assert.Equal(t, "https://example.com/oauth2/did:web:example.com:iam:holder", actual.String())
+	})
+	t.Run("ok", func(t *testing.T) {
+		webDID := did.MustParseDID("did:web:example.com:iam:holder")
+		actual, err := createOAuth2EndpointURL(webDID, "something")
+
+		require.NoError(t, err)
+		require.NotNil(t, actual)
+		assert.Equal(t, "https://example.com/oauth2/did:web:example.com:iam:holder/something", actual.String())
+	})
+	t.Run("nested path", func(t *testing.T) {
+		webDID := did.MustParseDID("did:web:example.com:iam:holder")
+		actual, err := createOAuth2EndpointURL(webDID, "///foo//bar")
+
+		require.NoError(t, err)
+		require.NotNil(t, actual)
+		assert.Equal(t, "https://example.com/oauth2/did:web:example.com:iam:holder/foo/bar", actual.String())
+	})
+	t.Run("did:web with port", func(t *testing.T) {
+		const didAsString = "did:web:example.com%3A8080:iam:holder"
+		webDID := did.MustParseDID(didAsString)
+
+		actual, err := createOAuth2EndpointURL(webDID, "something")
+
+		require.NoError(t, err)
+		require.NotNil(t, actual)
+		assert.Equal(t, "https://example.com:8080/oauth2/did:web:example.com%253A8080:iam:holder/something", actual.String())
```

Comment:
Since you fiddled with rawpath, what happens if I put this return path in a Location header. Is the framework going to escape it again or not?